prov/rxd/src/rxd_cq.c

```diff
@@ -52,15 +52,15 @@ static const char *rxd_cq_strerror(struct fid_cq *cq_fid, int prov_errno,
 
 	cq = container_of(cq_fid, struct rxd_cq, util_cq.cq_fid);
 
-	ofi_genlock_lock(&cq->util_cq.ep_list_lock);
+	ofi_mutex_lock(&cq->util_cq.ep_list_lock);
```

Comment:
The genlock vs mutex implementation is exactly the same. Switching back to the mutex lock shouldn't impact performance. I'm not convinced this is the source of the regression you're seeing.

Reply:
I question that the test results are reproducible enough to determine a 20ns difference.  The mutex calls go direct to pthread calls, the genlock calls access those indirectly via a function pointer.  The point of genlock is that it allows disabling the lock at runtime, which isn't possible calling mutex directly.

Reply:
Please reproduce the testing on your end.  In the slack channel, I linked my run command for the reproducer.

Reply:
From my test results:
```
Bisect 2 84a6bde5e -> main(for SHM)

84a6bde5e: .46  99f87709c:.46  <Seths OFI genlock changes in here>    318b3f49f: .45  55f7fd1ad: .45    6b62f34ea:.45   90baec9d3: .45   803af8801: .45   64b31bea3: .48   967f91458:.48    52a4364d9: .48   422241636:.48          main: .48
```

And running with this branch on main brings us from .48 to .46.